dbt_style_guide.md

```diff
@@ -1,237 +1,510 @@
+  Example:
+  ```sql
+  sum(
+      case when order_status = 'complete' then order_total end
+  ) as monthly_total,
 
-    select * from {{ ref('some_cte') }}
 
-),
+  {{ get_windowed_values(
+        strategy='sum',
+        partition='order_id',
+        order_by='created_at',
+        column_list=[
+            'final_cost'
+        ]
+  ) }} as total_final_cost
+  ```
 
-some_cte_agg as (
+- Our convention is to use all lowercase unless a specific scenario needs us to
+  do otherwise. This means that keywords, field names, function names, and file names
+  should all be lowercased.
 
-    select
-        id,
-        sum(field_4) as total_field_4,
-        max(field_5) as max_field_5
+- The `as` keyword should be used when aliasing a field or table
 
-    from some_cte
-    group by 1
+- Fields should be stated before aggregates / window functions
 
-),
+- Aggregations should be executed as early as possible before joining to another table.
 
-final as (
+- Ordering and grouping by a number (eg. group by 1, 2) is preferred over listing the column names (see [this rant](https://blog.getdbt.com/write-better-sql-a-defense-of-group-by-1/) for why). Note that if you are grouping by more than a few columns, it may be worth revisiting your model design. If you really need to, the [dbt_utils.group_by](https://github.com/dbt-labs/dbt-utils/tree/0.8.6/macros/sql/groupby.sql) function may come in handy.
 
-    select [distinct]
-        my_data.field_1,
-        my_data.field_2,
-        my_data.field_3,
+- Prefer `union all` to `union` [*](http://docs.aws.amazon.com/redshift/latest/dg/c_example_unionall_query.html)
 
-        -- use line breaks to visually separate calculations into blocks
-        case
-            when my_data.cancellation_date is null
-                and my_data.expiration_date is not null
-                then expiration_date
-            when my_data.cancellation_date is null
-                then my_data.start_date + 7
-            else my_data.cancellation_date
-        end as cancellation_date,
+- Avoid table aliases in join conditions (especially initialisms) – it's harder to understand what the table called "c" is compared to "customers".
 
-        some_cte_agg.total_field_4,
-        some_cte_agg.max_field_5
+- If joining two or more tables, _always_ prefix your column names with the table alias. If only selecting from one table, prefixes are not needed.
 
-    from my_data
-    left join some_cte_agg  
-        on my_data.id = some_cte_agg.id
-    where my_data.field_1 = 'abc'
-        and (
-            my_data.field_2 = 'def' or
-            my_data.field_2 = 'ghi'
-        )
-    having count(*) > 1
+- Be explicit about your join (i.e. write `inner join` instead of `join`). `left joins` are normally the most common, `right joins` often indicate that you should change which table you select `from` and which one you `join` to.
+
+
+Example:

```

Comment:
Is this Example in the wrong spot? I'd think it'd make more sense to put under line 237 'When dealing with long `when` or `where` clauses, predicates should be on a new line and indented.

Reply:
Not sure why it shows it in the spot it does here - I pushed a change earlier that shows it [here](https://github.com/dbt-labs/corp/blob/update/proserv_style_guide_changes/dbt_style_guide.md). It's possible you got this one in before I changed it! Either way, it's fixed. Good catch! =)

Reply:
Ha, most likely!